core/src/main/java/org/bitcoinj/wallet/Wallet.java

```diff
@@ -3710,7 +3708,7 @@ public Coin getBalance(BalanceType balanceType) {
         try {
             if (balanceType == BalanceType.AVAILABLE || balanceType == BalanceType.AVAILABLE_SPENDABLE) {
                 List<TransactionOutput> candidates = calculateAllSpendCandidates(true, balanceType == BalanceType.AVAILABLE_SPENDABLE);
-                CoinSelection selection = coinSelector.select(NetworkParameters.MAX_MONEY, candidates);
+                CoinSelection selection = DefaultCoinSelector.get().select(NetworkParameters.MAX_MONEY, candidates);
```

Comment:
I would consider having a `protected final` default coin selector to allow subclasses to override, rather than hardcoding `DefaultCoinSelector`.

Reply:
Reverted this line and the similar ones.